File: webhole/packages/python-flask/cd.py
def cd(path, old_path=None):
    import os
    if old_path is not None:
        try:
            os.chdir(old_path)
        except OSError:
            #print("❌ Error: Could not change to old path!")
            return old_path
    try:
        os.chdir(path)
        cwd = os.getcwd()
        # Normalize slashes and ensure proper Windows drive letter formatting
        cwd = os.path.normpath(cwd)
        if len(cwd) == 3 and cwd[1] == ":":
            cwd += "\\"  # Ensure "C:" becomes "C:\"
        return cwd
    except OSError:
        #print("❌ Error: Directory not found or permission denied!")
        return old_path
print(cd('__path__'))

File: webhole/packages/php/cd.php
<?php
function cd($path, $old_path = null) {
    if ($old_path !== null) {
        if (!@chdir($old_path)) {
            return "Error: Could not change to old path!";
        }
    }

    if (@chdir($path)) {
        $cwd = getcwd();

        // Normalize slashes and ensure the path is properly formatted
        $cwd = preg_replace('/[\/\\\\]+/', DIRECTORY_SEPARATOR, $cwd);

        // Fix Windows drive letter issue (e.g., "C:xampp" issue)
        if (strpos($cwd, ':') === 1) { // If there's a missing backslash after drive letter
            $cwd = strtoupper($cwd[0]) . ":\\" . substr($cwd, 2);
        }

        return $cwd;
    } else {
        return "Error: Directory not found or permission denied!";
    }
}

echo cd('__path__');